source/ceylon/time/internal/GregorianDate.ceylon

```diff
@@ -94,8 +94,7 @@ shared class GregorianDate( Integer dayOfEra )
         if ( years == 0 ) {
             return this;
         }
-        value newDay = day == 29 && month == february then 28 else day;
```

Comment:
I think that the only line that was wrong with previous version was this one. In fact, it explicitly sets date to 28 for february, which is why we get that bug.